### PR TITLE
done with week 2 assessment

### PR DESCRIPTION
**Pull request by gitgood-ENT**

### CODE REVIEW

File: code-challenges.test.js [hunk 1]
```diff
@@ -1,7 +1,9 @@
 // ASSESSMENT 2: Coding Practical Questions with Jest
 
+const { expect } = require("@jest/globals")
+
 // Please read all questions thoroughly
-// Pseudo coding is REQUIRED
+// Pseudo coding is REQUIRED  ---------- also on the expect block of the test
```
Comment (sjproctor) [inline code-challenges.test.js:6]: Tests describe expected behavior while pseudo coding describes processes.

File: code-challenges.test.js [hunk 2]
```diff
@@ -23,10 +25,47 @@ var num3 = -7
...
+//         return `${num} is not divisible by three`
+//     }
+// }
+
+// got red saying that  
+//      "● Test suite failed to run
+//               Your test suite must contain at least one test."
+//
+// Checked initial repo on github and found a difference in code on lines 4 and 5. The code that is there was not in the github repo and once commented out allowed my test to return green. Am curious as to why it happened.
+
```
Comment (sjproctor) [inline code-challenges.test.js:59]: VSCode adds unnecessary imports to the code file which prevent the tests from passing. they just need to be removed.

File: code-challenges.test.js [hunk 3]
```diff
@@ -23,10 +25,47 @@ var num3 = -7
...
+// Got green
+// Don't know how to refactor, actually, I could use a ternary
+
+const divByThree = (num) => {
+// if the number is divisible by three return it is divisble by three otherwise that it is not
+  return num % 3 === 0 ? `${num} is divisible by three` : `${num} is not divisible by three`
+}
+
```
Comment (sjproctor) [inline code-challenges.test.js:67]: Looks great!

File: code-challenges.test.js [hunk 4]
```diff
@@ -37,10 +76,44 @@ var randomNouns1 = ["streetlamp", "potato", "teeth", "conclusion", "nephew"]
...
+// declare the function capFirst that capitalizes the first letter of each string in an array
+const capFirst = (array) => {
+    // return the input array using .map to iterate over the array and return a new array
+    return array.map(firstUpper)
+}
+//Got green, moving to refactor => is it possible to combine the two functions into one?
+//Declare function that takes the array in and then uses map combined with the return from firstUpper to output expected array.
+//I think anything I doto combine them will make it longer or harder to read
+
```
Comment (sjproctor) [inline code-challenges.test.js:116]: Breaking the functions out is fine. But they can absolutely be combined into one, very readable, function.

File: code-challenges.test.js [hunk 5]
```diff
@@ -54,6 +127,52 @@ var vowelTester2 = "academy"
...
+    //use a for loop to iterate through the string and return the index of the first vowel
+    for (let i = 0; i < string.length; i++) {
+        //Use an if statement combined with isVowel to determine if the letter at the current index is a Vowel, and if it is log it/return it
+        if ((isVowel(string[i]))) {
+            console.log(i)
+            return i
+        }
+        
+    }
+}
```
Comment (sjproctor) [inline code-challenges.test.js:178]: Interesting approach. I think some logical operators might shorten your code significantly.

File: coding-concepts.js [hunk 6]
```diff
@@ -8,40 +8,59 @@
...
 // a) Your answer:
-// b) Verify and explain:
+
+// This will log the string "Echo 2021" in the form of an array with each letter being an item(string) in the array
+//['E', 'c', 'h', 'o', ' ', '2', '0', '2', '1']
+
+
+// b) Verify and explain: 
+
+// Verified; it does this because .split is splitting the string at each item in the string because it is splitting at every "", and there is nothing between the quote marks so it splits after every item
```
Comment (sjproctor) [inline coding-concepts.js:21]: Nice!

File: coding-concepts.js [hunk 7]
```diff
@@ -8,40 +8,59 @@
...
 const greeter = (name) => {
   `Hello, ${name}!`
 }
-// console.log(greeter("LEARN Student"))
+console.log(greeter("LEARN Student"))
 
 // a) Your answer:
+
+// undefined; there is no return in the function
+
```
Comment (sjproctor) [inline coding-concepts.js:34]: Good catch!

File: interview-questions.md [hunk 8]
```diff
@@ -8,62 +8,70 @@ Then, research the question to expand on your answer. Even if you feel you have
...
 
-  Researched answer:
+  Researched answer: A class is a template for objects, the class will define the objects properties, including range and default values, and it will define the objects behaviour. The object is an instance of a class and has values for its properties.
 
 
 
 6. STRETCH: What is hoisting in JavaScript?
 
-  Your answer:
+  Your answer: Hoisting is when a function or variable is decfined in the code, but it can be called on before it was defined in the code. i.e. a function can be defined on like 234 but it can be called and used in line 123. const will be hoisted, but var and let will not be. Variables that are not assigned as const var or let will be automatically assigned to the global scale and will be hoisted as well.
```
Comment (sjproctor) [inline interview-questions.md:59]: Nice job.

**Review by sjproctor — APPROVED** [review]
Great work on the code concepts and the interview questions! I appreciate the work you put in. Your code logic looks good, I am glad you are exploring different approaches to the logic.

File: code-challenges.test.js (hunk 5 above)
Comment (gitgood-ENT) [reply code-challenges.test.js:178]: I agree, I came across the following recently and i think it cleans it up nicely.

const vowelTester = (string) => {
  return string.search(/[aeiou]/gi)
}

Comment (sjproctor) [reply code-challenges.test.js:178]: Yeah that is a library called Regular Expressions. It is a set of string methods. That works. There are also shorter programmatic solutions.

Comment (gitgood-ENT) [reply code-challenges.test.js:178]: You've caught my attention, what are the shorter solutions?

**Pull request merged**